src/plugins/data_source/server/saved_objects/credential_saved_objects_client_wrapper.ts

```diff
@@ -0,0 +1,278 @@
+      options: SavedObjectsUpdateOptions = {}
+    ): Promise<SavedObjectsUpdateResponse<T>> => {
+      this.validateType(type);
+      this.validateAttributes(attributes);
+
+      const { title, authType, credentialMaterials, description } = attributes;
+
+      if (this._noAuthType === credentialMaterials.credentialMaterialsType) {
+        return await wrapperOptions.client.create(type, attributes, options);
+      }
+
+      const encryptedAttributes: Partial<T> = {
+        title,
+        authType,
+        credentialMaterials: await this.validateAndEncryptCredentialMaterials(credentialMaterials),
+        description,
+      };
+
+      return await wrapperOptions.client.update(type, id, encryptedAttributes, options);
+    };
+
+    const bulkUpdateWithCredentialMaterialsEncryption = async <T = unknown>(
+      objects: Array<SavedObjectsBulkUpdateObject<T>>,
+      options?: SavedObjectsBulkUpdateOptions
+    ): Promise<SavedObjectsBulkUpdateResponse<T>> => {
+      objects = await Promise.all(
+        objects.map(async (object) => {
+          const { type, attributes } = object;
+
+          this.validateType(type);
+          this.validateAttributes(attributes);
+
+          const { title, authType, credentialMaterials, description } = attributes;
+
+          if (this._noAuthType === credentialMaterials.credentialMaterialsType) {
+            return object;
+          }
+
+          const encryptedAttributes = {
+            title,
+            authType,
+            credentialMaterials: await this.validateAndEncryptCredentialMaterials(
+              credentialMaterials
+            ),
+            description,
+          };
+
+          return ({
+            ...object,
+            attributes: encryptedAttributes,
+            // Unfortunately this throws a typescript error without the casting.  I think it's due to the
+            // convoluted way SavedObjects are created.
+          } as unknown) as SavedObjectsBulkUpdateObject<T>;
+        })
+      );
+
+      return await wrapperOptions.client.bulkUpdate(objects, options);
+    };
+
+    return {
+      ...wrapperOptions.client,
+      create: createWithCredentialMaterialsEncryption,
+      bulkCreate: bulkCreateWithCredentialMaterialsEncryption,
+      checkConflicts: wrapperOptions.client.checkConflicts,
+      delete: wrapperOptions.client.delete,
+      find: wrapperOptions.client.find,
+      bulkGet: wrapperOptions.client.bulkGet,
+      get: wrapperOptions.client.get,
+      update: updateWithCredentialMaterialsEncryption,
+      bulkUpdate: bulkUpdateWithCredentialMaterialsEncryption,
+      errors: wrapperOptions.client.errors,
+      addToNamespaces: wrapperOptions.client.addToNamespaces,
+      deleteFromNamespaces: wrapperOptions.client.deleteFromNamespaces,
+    };
+  };
+
+  private validateType(type: string) {
+    if (this._type !== type) {
+      throw SavedObjectsErrorHelpers.createUnsupportedTypeError(type);
+    }
+    return;
+  }
```

Comment:
why throw error ? if you throw error here, the saved object won't be stored, thus you cause error for all saved objects except credentials

Reply:
Good catch. Will only validate on credential type and call default contract otherwise. I didn't intend to support encryption for all types using the factory pattern because deciding which fields to encrypt is not scalable. Open to more ideas.